### PR TITLE
build(webpack): Upgrade to `webpack@5`

### PR DESCRIPTION
**Pull request by billyvg**
Upgrade to `webpack@5`

### CODE REVIEW

**Comment by billyvg** [comment]
I believe this is due to `react-bootstrap` import change
image

File: build-utils/optional-locale-chunk-plugin.js [hunk 1]
```diff
@@ -10,18 +10,24 @@
...
  * *without* all locale chunks loaded.
+ *
+ * Note that the locale actually gets initially loaded in the server template and
+ * the frontend code (in `app/translations`) may be redundant.
  */
 const PLUGIN_NAME = 'OptionalLocaleChunkPlugin';
 
-const clearLocaleChunks = chunks =>
-  chunks
+const clearLocaleChunks = chunks => {
```
Comment (billyvg) [inline build-utils/optional-locale-chunk-plugin.js:19]: Intentionally did this so it's a bit more clear that we're not expecting a return value.

File: build-utils/optional-locale-chunk-plugin.js [hunk 2]
```diff
@@ -10,18 +10,24 @@
...
+  Array.from(chunks)
```
Comment (billyvg) [inline build-utils/optional-locale-chunk-plugin.js:20]: `chunks` is no longer an array in 5.

File: package.json [hunk 3]
```diff
@@ -61,12 +61,13 @@
...
     "clean-webpack-plugin": "^3.0.0",
     "color": "^3.1.3",
-    "compression-webpack-plugin": "^6.1.1",
+    "compression-webpack-plugin": "7.0.0",
     "copy-text-to-clipboard": "2.2.0",
     "core-js": "^3.2.1",
     "create-react-class": "^15.6.2",
     "crypto-js": "4.0.0",
     "css-loader": "^5.0.1",
+    "css-minimizer-webpack-plugin": "^1.3.0",
```
Comment (billyvg) [inline package.json:70]: Replaces `optimize-css-assets-webpack-plugin`

File: package.json [hunk 4]
```diff
@@ -180,13 +182,21 @@
...
-    "**/rrweb": "^0.7.33"
+    "**/rrweb": "^0.7.33",
+    "webpack": "5.27.2",
+    "css-loader": "^5.0.0",
+    "dotenv-webpack": "^6.0.0",
+    "html-webpack-plugin": "^5.0.0",
+    "less-loader": "^8.0.0",
+    "style-loader": "^2.0.0",
+    "terser-webpack-plugin": "^5.0.0",
+    "webpack-virtual-modules": "^0.4.2"
```
Comment (billyvg) [inline package.json:199]: This is because of storybook :(

File: webpack.config.js [hunk 5]
```diff
@@ -373,27 +373,38 @@ let appConfig = {
       ),
     },
 
+    fallback: {
+      vm: require.resolve('vm-browserify'),
+      stream: require.resolve('stream-browserify'),
+      crypto: require.resolve('crypto-browserify'),
```
Comment (billyvg) [inline webpack.config.js:379]: Need to wait for a new `eslint-import-resolver-webpack` release in order to get rid of these (I think): https://github.com/benmosher/eslint-plugin-import/issues/631

File: build-utils/optional-locale-chunk-plugin.js (hunk 2 above)
Comment (EvanPurkhiser) [reply build-utils/optional-locale-chunk-plugin.js:20]: How many tacos do I have to pay you to convert our webpack to typescript. It's probably about time

File: webpack.config.js [hunk 6]
```diff
@@ -373,27 +374,42 @@ let appConfig = {
...
     modules: ['node_modules'],
     extensions: ['.jsx', '.js', '.json', '.ts', '.tsx', '.less'],
   },
   output: {
     path: distPath,
     filename: '[name].js',
-
-    // Rename global that is used to async load chunks
-    // Avoids 3rd party js from overwriting the default name (webpackJsonp)
-    jsonpFunction: 'sntryWpJsonp',
```
Comment (EvanPurkhiser) [inline webpack.config.js:385]: Bumped into this the other day and was pretty sure it wasn't used. Good to see it removed

File: webpack.config.js [hunk 7]
```diff
@@ -373,27 +374,42 @@ let appConfig = {
...
+      // django template.
+      chunks: 'async',
       maxInitialRequests: 5,
       maxAsyncRequests: 7,
       cacheGroups,
     },
+
+    // This only runs in production mode
+    // Grabbed this example from https://github.com/webpack-contrib/css-minimizer-webpack-plugin
+    minimizer: ['...', new CssMinimizerPlugin()],
```
Comment (EvanPurkhiser) [inline webpack.config.js:410]: damn, this `'...'` thing is magic

**Review by evanpurkhiser — APPROVED** [review]
Great work on this billy, it was nice and easy to review. ⭐⭐⭐⭐⭐ would review again

File: webpack.config.js (hunk 6 above)
Comment (billyvg) [reply webpack.config.js:385]: It was used fwiw - it was added after Stripe overwrote the `webpackJsonp` global and broke our SPA, but this was removed in webpack5 where it defaults to `package.json -> name`

File: webpack.config.js (hunk 7 above)
Comment (billyvg) [reply webpack.config.js:410]: yea magic strings :|

File: build-utils/optional-locale-chunk-plugin.js (hunk 2 above)
Comment (billyvg) [reply build-utils/optional-locale-chunk-plugin.js:20]: 😬

**Pull request merged**